Content.Shared/Examine/ExamineSystemShared.cs

```diff
@@ -214,6 +214,9 @@ public bool InRangeUnOccluded<TState>(MapCoordinates origin, MapCoordinates othe
 
         public bool InRangeUnOccluded(EntityUid origin, EntityUid other, float range = ExamineRange, Ignored? predicate = null, bool ignoreInsideBlocker = true)
         {
+            if (!_containerSystem.IsInSameOrTransparentContainer(origin, other, userSeeInsideSelf: true))
+                return false;
+
```

Comment:
This should be a separate check as it's a very hot path, just put it before inrangeunoccluded on the flash check.

Reply:
Thanks for the note regarding the hot path. So I reverted the last commit since this is how it was done previously.